account_bank_statement_import_bypass_check/wizard/account_bank_statement_import.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import logging
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountBankStatementImport(models.TransientModel):
+    _inherit = "account.bank.statement.import"
+
+    check_journal = fields.Boolean(default=True)
+    check_message = fields.Char(readonly=True)
+    force_journal = fields.Boolean(
+        string="Do you want to force this import ?", default=False
+    )
+
+    @api.multi
+    def import_file(self):
+        """ Override to make the import in 2 step.
+
+        To provide to user possiblity to bypass the journal check.
+        """
+        self.ensure_one()
+        currency_code, account_number, stmts_vals = self.with_context(
+            active_id=self.ids[0]
+        )._parse_file(base64.b64decode(self.data_file))
+        if self.check_journal:
+            try:
+                self._find_additional_data(currency_code, account_number)
+            except UserError as u_error:
+                if not self.check_journal:
```

Comment:
Are you sure of this test?
If your test line 32 is true, your test here can't be true.
`self.check_journal != not self.check_journal`

Or call of `_find_additional_data` change the value of `check_journal`.
In this case, can you add a little comment to explain this?

Reply:
New comment on https://github.com/OCA/bank-statement-import/pull/243/files#diff-204ab1d34e5796f546481c0211664ef5R35-R36